pkg/swagger/generator/model.go

```diff
@@ -161,73 +155,102 @@ func makeGenDefinitionHierarchy(name, pkg, container string, schema spec.Schema,
 		GenSchema:    pg.GenSchema,
 		DependsOn:    pg.Dependencies,
 		ExtraSchemas: gatherExtraSchemas(pg.ExtraSchemas),
-		Imports:      imports,
+		Imports:      collectSortedImports(pg.GenSchema),
 	}, nil
 }
 
-type importOpts struct {
-	pkg  string
-	name string
+type importStmt struct {
+	ImportPath string
+	AsName     string
+	MustAsName bool
 }
 
-func findImports(sch *GenSchema, opts *importOpts) map[string]string {
-	imp := map[string]string{}
-	if sch.Name != opts.name {
-		if sch.Pkg != opts.pkg {
-			// not in the same pkg, import pkg
-			if sch.Pkg != "" && sch.Module != "" {
-				imp[sch.Module] = sch.Pkg
-			}
-			// on cross pkg import, add pkg alias to type
-			if sch.PkgAlias != "" {
-				sch.KclType = sch.PkgAlias + "." + sch.KclType
+func collectSortedImports(model GenSchema) []importStmt {
+	importMap := map[string]importStmt{}
+	collectImports(&model, model.Pkg, importMap)
+	sortedPkgPaths := make([]string, 0, len(importMap))
+	sortedImports := make([]importStmt, 0, len(importMap))
+	for k := range importMap {
+		sortedPkgPaths = append(sortedPkgPaths, k)
+	}
+	sort.Strings(sortedPkgPaths)
+	for _, k := range sortedPkgPaths {
+		sortedImports = append(sortedImports, importMap[k])
+	}
+	return sortedImports
+}
+
+func getImportAsName(imp map[string]importStmt, sch *GenSchema) string {
+	parts := strings.Split(sch.Pkg, ".")
+	asName := ""
+	for i := len(parts) - 1; i >= 0; i-- {
+		conflict := false
+		// when conflict with other import as name, the `import as` name will be "{parentPkgName}strings.Title({PkgAlias})"
+		asName = parts[i] + strings.ToTitle(asName)
+		for _, v := range imp {
+			if v.AsName == asName {
+				conflict = true
+				break
 			}
 		}
+		if !conflict {
+			return asName
+		}
 	}
-	if sch.Items != nil {
-		sub := findImports(sch.Items, opts)
-		for k, v := range sub {
-			imp[k] = v
+	mangledAsName := "kusionMangled" + strings.ToTitle(asName)
+	for _, v := range imp {
+		if v.AsName == asName {
+			log.Printf("[WARN] the import paths in module %s.%s are confict, please resolve it properly", sch.Pkg, sch.Module)
 		}
+	}
+	return mangledAsName
+}
+
+// findImports add import stmt from sch to opts if needed to the map of import stmts.
```

Comment:
s/findImports/collectImports

Besides, the comments of the function does not seem to match the usage of the function.

Reply:
fixed.